iothub/device/src/Transport/AmqpIoT/AmqpIoTErrorAdapter.cs

```diff
@@ -77,7 +76,7 @@ public static Exception ToIotHubClientContract(AmqpException amqpException)
             }
             else if (Equals(AmqpErrorCode.DecodeError, amqpSymbol))
             {
-                return new AmqpIoTResourceException(message, amqpException);
+                return new IotHubException(message, amqpException);
```

Comment:
Just wondering when does this happen? 

In the docs I see this - Mismatch between AMQP message sent by client and received by service; collect logs and contact service

What does mismatch mean here?

Reply:
It means that the data that the client encodes and sends over the amqp channel is not able to be decoded by the service (assuming that the same algorithm is used for encoding and decoding). This exception will likely originate from the amqp library directly, and not from the service.

Reply:
So does it always mean an error on client encoding?

Reply:
yes; this is one of the error codes that the amqp lib defines, but we don't expect to see this exception. Also, if we ever see this exception for one operation, we would see it for all operations as well (since this is an encoding-decoding based error code).